Signal/src/ViewControllers/Photos/PhotoLibrary.swift

```diff
@@ -244,6 +244,10 @@ class PhotoCollection {
 
         return PhotoCollectionContents(fetchResult: fetchResult, localizedTitle: localizedTitle())
     }
+    
+    var isEmpty: Bool {
+    	return collection.estimatedAssetCount == 0
```

Comment:
It's not clear to me that `estimatedAssetCount` is guaranteed to be accurate. What's your rationale?

Reply:
According to the documentation, it's indeed not guaranteed to always be up-to-date or it might not always be loaded in time:

> This count may not match the current number of assets in the collection. To get the most recent count, fetch the collection’s assets with the fetchAssets(in:options:) method and read the count property of the fetch result.
If asset count information is not available for the collection, this property’s value is NSNotFound

False positives wil hopefully self-correct in time, and false negatives might display e.g. an empty Favourites collection. It's not perfect, but it seemed like a good trade-off.

I tried using `fetchAssets(in:options:)` with a `limit` of `1`, but that was as slow as without limit.

The factor slowing this down is actually the number of albums I have in my Photo Library (automatically converted from iPhoto's Events)

Reply:
To make sure that I understand, the implication is that we might (rarely) show an empty collection in the album list. Is that right?

Have you tested what happens when the user gets in that state? (e.g. change the code to list empty collections). Can you send a screenshot of what the user sees in that case? And what happens when the users taps on that collection?

If all that's reasonable, then I'd agree that's a preferable exchange for a much snappier load time.

One small thing, apparently `estimatedAssetCount` is one of the things we can filter by in our query. No biggie, but it seems like it might be more to the point to just filter on that in our query rather than checking each one. What do you think?

https://developer.apple.com/documentation/photokit/phfetchoptions

Reply:
> To make sure that I understand, the implication is that we might (rarely) show an empty collection in the album list. Is that right?

Yes, and right after favouriting your first photo ever, Favourites might still be hidden until `estimatedAssetCount` catches up (same for shortly after adding the first photo to an empty album)

> Have you tested what happens when the user gets in that state? (e.g. change the code to list empty collections). Can you send a screenshot of what the user sees in that case?

![IMG_7771](https://user-images.githubusercontent.com/138759/62825090-93ca8880-bba6-11e9-91eb-3f0c9b91db9a.jpg)


> And what happens when the users taps on that collection?

Nothing happens, the currently selected photos remain the same.

 
> If all that's reasonable, then I'd agree that's a preferable exchange for a much snappier load time.

I think so too, but to lower the risks even more, you could not filter out the empty collections and add an empty state.

> One small thing, apparently `estimatedAssetCount` is one of the things we can filter by in our query. No biggie, but it seems like it might be more to the point to just filter on that in our query rather than checking each one. What do you think?
> 
> https://developer.apple.com/documentation/photokit/phfetchoptions

That's a possibility indeed, but I'd rather keep my changes as limited as possible, since this is my first PR ;-)

Reply:
I played around with this a bit and it seems like `estimatedAssetCount` always returns `NSNotFound` for smart albums, which means things like `time lapse`, `favorites`, etc. will always show up even if they're empty. I'm not sure if there is a good work around for this, but it'd be nice to investigate it a little further.

Reply:
Aha, thanks for testing. I'll try to find a better solution then when I'm back from my holidays.

Reply:
Awesome! Thanks so much for your contributions. I've incorporated all of your work, other than the `estimatedAssetCount` change (6ecc8c33f98204591d11628cfa65771af16deb91), into the upcoming 2.42 release.